adapter/PreciceInterface.c

```diff
@@ -530,18 +649,26 @@ void PreciceInterface_ConfigureTetraFaces( PreciceInterface * interface, Simulat
 
 void PreciceInterface_ConfigureCouplingData( PreciceInterface * interface, SimulationData * sim, InterfaceConfig const * config )
 {
-
 	interface->nodeScalarData = malloc( interface->numNodes * sizeof( double ) );
-	interface->nodeVectorData = malloc( interface->numNodes * 3 * sizeof( double ) );
+  interface->node2DScalarData = malloc( interface->num2DNodes * sizeof( double ));
+  interface->nodeVectorData = malloc( interface->numNodes * 3 * sizeof( double ) );
+  interface->node2DVectorData = malloc( interface->num2DNodes * 2 * sizeof( double ) );
+
+  int dim = interface->dim;
+  for (int i = 0; i < interface->num2DNodes; i++)
+  {
+    interface->node2DScalarData[i] = 0.0;
+    interface->node2DVectorData[i*dim] = 0.0;
+    interface->node2DVectorData[i*dim + 1] = 0.0;
+  }
+
```

Comment:
The 3D case fails here, since `interface->num2DNode` is uninitialized and you have essentially a memory leak. All '2D' needs to be conditional.